src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -71,75 +89,249 @@ const DotPlotPage = (props) => {
   const { experimentId } = props;
 
   const dispatch = useDispatch();
-  const config = useSelector((state) => state.componentConfig[plotUuid]?.config);
+
+  const {
+    config,
+    plotData,
+    loading: plotDataLoading,
+    error: plotDataError,
+  } = useSelector((state) => state.componentConfig[plotUuid]) || {};
+
   const {
     fetching: genesFetching,
     data: highestDispersionGenes,
   } = useSelector((state) => state.genes.properties.views[plotUuid] || {});
-  const cellSets = useSelector((state) => state.cellSets);
+
+  const cellSets = useSelector(getCellSets());
   const {
     loading: cellSetsLoading,
     error: cellSetsError,
-    hierarchy,
+    hierarchy: cellSetHierarcy,
+    properties: cellSetProperties,
   } = cellSets;
 
-  useEffect(() => {
-    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+  const [moreThanTwoGroups, setMoreThanTwoGroups] = useState(false);
 
-    if (hierarchy.length === 0) dispatch(loadCellSets(experimentId));
+  useEffect(() => {
+    if (!config) dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+    if (cellSetHierarcy.length === 0) dispatch(loadCellSets(experimentId));
   }, []);
 
+  const previousComparedConfig = useRef(null);
+  const getComparedConfig = (updatedConfig) => _.pick(
+    updatedConfig,
+    ['useMarkerGenes',
+      'nMarkerGenes',
+      'selectedGenes',
+      'selectedCellSet',
+      'selectedPoints'],
+  );
+
+  const hasGroupsToCompare = (baseCluster, filterCluster) => {
+    if (cellSetsLoading || !baseCluster || !filterCluster) return false;
+
+    // filterBy has the shape louvain/louvain-1
+    const [filterRootNode, filterKey] = filterCluster.split('/');
+
+    // If 'All" is chosen for the dropdown, there will always be representation from more than 1 group
+    if (filterRootNode === 'All') return true;
+
+    const filterClusterCellIds = Array.from(cellSetProperties[filterKey].cellIds);
+
+    const baseClusterParent = cellSetHierarcy.find((cellSet) => cellSet.key === baseCluster);
+    const baseClusterKeys = baseClusterParent.children.map((child) => child.key);
+    const baseClusterCellIds = baseClusterKeys.map((key) => cellSetProperties[key].cellIds);
+
+    // Check if there is at least 2 baseClusters with elements common to filterCluster
+    let numIntersections = 0;
+    return baseClusterCellIds.some((baseCellIds) => {
+      if (filterClusterCellIds.some((filterCellId) => baseCellIds.has(filterCellId))) {
+        numIntersections += 1;
+      }
+
+      return numIntersections === 2;
+    });
+  };
+
+  const hasGroups = useMemo(
+    () => hasGroupsToCompare(config?.selectedCellSet, config?.selectedPoints),
```

Comment:
Initially when the config is not defined we still do this calculation in the function, maybe its better to put a check in `hasGroupsToCompare` if config is defined?

Reply:
There is already a check inside `hasGroupsToCompare` to check for the existence of the inputs. I'll move it out into the `useMemo` function body